cli/src/commands/provider/gen.ts

```diff
@@ -15,29 +15,56 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+import { rm } from "fs/promises";
+
+import { color } from "@oclif/color";
+import { Flags } from "@oclif/core";
+
 import { BaseCommand, baseFlags } from "../../baseCommand.js";
 import { commandObj } from "../../lib/commandObj.js";
 import { ensureComputerPeerConfigs } from "../../lib/configs/project/provider.js";
 import {
   CHAIN_FLAGS,
   PROVIDER_CONFIG_FULL_FILE_NAME,
   PROVIDER_SECRETS_CONFIG_FULL_FILE_NAME,
+  DOT_FLUENCE_DIR_NAME,
+  SECRETS_DIR_NAME,
 } from "../../lib/const.js";
 import { initCli } from "../../lib/lifeCycle.js";
 import {
   getFluenceSecretsDir,
   ensureFluenceConfigsDir,
+  ensureProviderSecretsConfigPath,
 } from "../../lib/paths.js";
+import { confirm } from "../../lib/prompt.js";
 
 export default class Gen extends BaseCommand<typeof Gen> {
   static override description = `Generate Config.toml files according to ${PROVIDER_CONFIG_FULL_FILE_NAME} and secrets according to ${PROVIDER_SECRETS_CONFIG_FULL_FILE_NAME}`;
   static override examples = ["<%= config.bin %> <%= command.id %>"];
   static override flags = {
     ...baseFlags,
     ...CHAIN_FLAGS,
+    "reset-secrets": Flags.boolean({
+      description: `Remove nox secrets at ${DOT_FLUENCE_DIR_NAME}/${PROVIDER_SECRETS_CONFIG_FULL_FILE_NAME} and ${DOT_FLUENCE_DIR_NAME}/${SECRETS_DIR_NAME} to generate new ones`,
+      default: false,
+    }),
   };
   async run(): Promise<void> {
-    await initCli(this, await this.parse(Gen));
+    const { flags } = await initCli(this, await this.parse(Gen));
+    const providerSecretsConfigPath = await ensureProviderSecretsConfigPath();
+    const fluenceSecretsDir = getFluenceSecretsDir();
+
+    if (
+      flags["reset-secrets"] &&
+      (await confirm({
+        message: `Are you sure you want to remove ${color.yellow(providerSecretsConfigPath)} and ${color.yellow(fluenceSecretsDir)} to generate new ones`,
+        default: true,
```

Comment:
Maybe it should be `false`?

Reply:
if default is false then there will be no way to run this non-interactively at all if it's really needed for some reason (e.g. in tests)

Reply:
Can we use env here?

Reply:
how?

Reply:
ah, also that "default" is only in regards of what happens in non-interactive mode by default. The actual default value that you can just press enter is still YES